conductor_api/src/conductor/base.rs

```diff
@@ -320,14 +322,63 @@ impl Conductor {
         Ok(())
     }
 
+    pub fn start_instance(&mut self, id: &String) -> Result<(), HolochainInstanceError> {
+        let mut instance = self.instances.get(id)?.write().unwrap();
+        notify(format!("Starting instance \"{}\"...", id));
+
+        // Get instance DNA so we can read out required bridge definitions:
+        let dna =
+            instance
+                .state()?
+                .nucleus()
+                .dna()
+                .ok_or(HolochainInstanceError::InternalFailure(
+                    HolochainError::DnaMissing,
+                ))?;
+
+        // Make sure required bridges are configured and started:
+        for zome in dna.zomes.values() {
+            for bridge in zome.bridges.iter() {
+                if bridge.presence == BridgePresence::Required {
+                    let handle = bridge.handle.clone();
+                    let bridge_config = self
+                        .config
+                        .bridges
+                        .iter()
+                        .find(|b| b.handle == handle)
+                        .ok_or(HolochainInstanceError::RequiredBridgeMissing(
+                            handle.clone(),
+                        ))?;
+                    self.instances
+                        .get(&bridge_config.callee_id)
+                        .ok_or(HolochainInstanceError::RequiredBridgeMissing(
+                            handle.clone(),
+                        ))?
+                        .read()
+                        .unwrap()
+                        .active()
+                        .ok_or(HolochainInstanceError::RequiredBridgeMissing(handle))?;
+                }
+            }
+        }
+        instance.start()
+    }
+
+    pub fn stop_instance(&mut self, id: &String) -> Result<(), HolochainInstanceError> {
+        let instance = self.instances.get(id)?;
+        notify(format!("Stopping instance \"{}\"...", id));
+        instance.write().unwrap().stop()
+    }
+
     /// Starts all instances
     pub fn start_all_instances(&mut self) -> Result<(), HolochainInstanceError> {
-        self.instances
-            .iter_mut()
-            .map(|(id, hc)| {
-                notify(format!("Starting instance \"{}\"...", id));
-                hc.write().unwrap().start()
-            })
+        self.config
+            .instances
+            .iter()
+            .map(|instance_config| instance_config.id.clone())
+            .collect::<Vec<String>>()
+            .iter()
```

Comment:
hmmm do we have to collect and call another iter()? Can it continue from the previous iter?

Reply:
Yes, we actually have to because we are borrowing `self` in the first line and then again in the second `map`. The `collect()` and `iter()` ends the first borrow and enables the second one.